File: Z-VocEngl/ZVocEngl.DAL/Data/Models/WordInformation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace ZVocEngl.DAL.Data.Models
{
    [Table("WordInformations")]
    public class WordInformation
    {
        public int Id { get; set; }
        public string Text { get; set; }
        public int WordId { get; set; }
        public int PartOfSpeechId { get; set; }
        public int TypeId { get; set; }

        // nav
        public ICollection<Example> Examples { get; set; }

        public ICollection<DefinitionSynonym> DefinitionSynonyms { get; set; }
        public ICollection<CollocationWord> CollocationWords { get; set; }
        public Type Type { get; set; }
        public Word Word { get; set; }
        public PartsOfSpeech PartOfSpeech { get; set; }
    }
}

File: z-vocengl-client/src/app/models/word-information.ts
import {Example} from './example';
import {Type} from './type';
import {Word} from './word';
import {PartOfSpeech} from './part-of-speech';
import {DefinitionSynonym} from './definition-synonym';
import {CollocationWord} from './collocation-word';

export class WordInformation {
  public id: number;
  public text: string;
  public wordId: number;
  public partOfSpeechId: number;
  public typeId: number;
  // nav
  public examples: Example[];
  public definitionsSynonyms: DefinitionSynonym[];
  public collocationWords: CollocationWord[];
  public type: Type;
  public word: Word;
  public partOfSpeech: PartOfSpeech;
}
